### PR TITLE
document the no-deps design constraints

### PR DESCRIPTION
**Pull request by iamqizhao**
replace #345

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ Prerequisites
 
 This requires Go 1.4 or above.
 
+Constraints
+-----------
+The grpc package should only depend on standard Golang packages and a short list of exceptions. A new addition to the list requires a discussion with gRPC-Go authors and consultants.
```
Comment (dsymonds) [inline README.md]: s/Golang/Go/

**Comment by philips** [comment]
I still feel we must say explicitly what that list of exceptions is if we want the community to help out.

**Comment by iamqizhao** [comment]
@philips 

I think we have made it clear that it is not strict constraint. The short list could be dynamic and we could accept new ones via discussion (which is required).

File: README.md (hunk 1 above)
Comment (iamqizhao) [inline README.md]: @dsymonds  done.

**Comment by philips** [comment]
@iamqizhao We can just link to this list and say you can only rely on those non-builtin packages: http://godoc.org/google.golang.org/grpc?imports

**Comment by dsymonds** [comment]
That's not true. Using an existing dependency is an easy case, but we aren't forbidding new dependencies completely; if they are justified then they're okay.

**Comment by philips** [comment]
@dsymonds I am trying to find some way of pointing at this list mentioned in the README now: "A new addition to the list requires a discussion with gRPC-Go authors and consultants." Isn't this link that list? http://godoc.org/google.golang.org/grpc?imports

**Comment by philips** [comment]
I am trying to make it non-ambiguous so we know when we have to have the discussion or not with new contributors.

**Comment by dsymonds** [comment]
Ah, I see what you're getting at. Yes, that's probably okay to link to for this purpose.

(FYI, this applies not just to new contributors; @iamqizhao would have to justify new dependencies too.)

**Comment by iamqizhao** [comment]
Definitely. Both first-party and third-party contributors are required to justify if they bring new dependencies to gRPC-Go. I can have a PR to put this list in if it is helpful.

**Comment by iamqizhao** [comment]
addressed in #378

**Pull request merged**